src/app/products/charges/create-charge/create-charge.component.ts

```diff
@@ -0,0 +1,221 @@
+          break;
+        case 3:
+          this.chargeCalculationTypeData = this.chargesTemplateData.clientChargeCalculationTypeOptions;
+          this.chargeTimeTypeData = this.chargesTemplateData.clientChargeTimeTypeOptions;
+          break;
+        case 4:
+          this.chargeCalculationTypeData = this.chargesTemplateData.shareChargeCalculationTypeOptions;
+          this.chargeTimeTypeData = this.chargesTemplateData.shareChargeTimeTypeOptions;
+          break;
+      }
+    });
+  }
+
+  /**
+   * @returns {any} Filtered charge calculation type data.
+   */
+  filteredChargeCalculationType(): any {
+    return this.chargeCalculationTypeData.filter((chargeCalculationType: any) => {
+      if (this.chargeForm.get('chargeTimeType').value === 12 && (chargeCalculationType.id === 3 || chargeCalculationType.id === 4)) {
+        return false;
+      }
+      if (this.chargeForm.get('chargeTimeType').value !== 12 && chargeCalculationType.id === 5) {
+        return false;
+      }
+      if (this.chargeForm.get('chargeAppliesTo').value === 2) {
+        if (!(this.chargeForm.get('chargeTimeType').value === 5 || this.chargeForm.get('chargeTimeType').value === 16) && chargeCalculationType.id === 2) {
+          return false;
+        }
+      }
+      return true;
+    });
+  }
+
+  /**
+   * Sets the conditional controls of the user form
+   */
+  setConditionalControls() {
+    this.chargeForm.get('chargeAppliesTo').valueChanges.subscribe((chargeAppliesTo) => {
+      this.chargeForm.get('penalty').enable();
+      switch (chargeAppliesTo) {
+        case 1: // Loan
+          this.chargeForm.addControl('chargePaymentMode', new FormControl('', Validators.required));
+          this.chargeForm.removeControl('incomeAccountId');
+          break;
+        case 2: // Savings
+          this.chargeForm.removeControl('chargePaymentMode');
+          this.chargeForm.removeControl('incomeAccountId');
+          break;
+        case 3: // Client
+          this.chargeForm.removeControl('chargePaymentMode');
+          this.chargeForm.addControl('incomeAccountId', new FormControl(''));
+          break;
+        case 4: // Shares
+          this.chargeForm.removeControl('chargePaymentMode');
+          this.chargeForm.removeControl('incomeAccountId');
+          this.chargeForm.get('penalty').setValue(false);
+          this.chargeForm.get('penalty').disable();
+          break;
+      }
+      this.chargeForm.get('chargeCalculationType').reset();
+      this.chargeForm.get('chargeTimeType').reset();
+    });
+    this.chargeForm.get('chargeTimeType').valueChanges.subscribe((chargeTimeType) => {
+      this.chargeForm.removeControl('feeFrequency');
+      this.chargeForm.removeControl('feeInterval');
+      this.chargeForm.removeControl('feeOnMonthDay');
+      this.chargeForm.removeControl('addFeeFrequency');
+      if (this.chargeForm.get('chargeAppliesTo').value !== 4) {
+        this.chargeForm.get('penalty').enable();
+      }
```

Comment:
Isn't this taken care of above by subscribing to `chargeAppliesTo`. Why do we need this here?

Reply:
It is because if we select chargeAppliesTo as 4, and then change the chargeTimeType, the penalty will re-enable without the if statement (We want the chargeTimeType to always be disabled for 4 regardless of what the chargeTimeType is).

Reply:
ok